app/js/app/directives/equation_editor/EquationInput.js

```diff
@@ -11,7 +11,7 @@ define([], function() {
             restrict: "A",
             templateUrl: "/partials/equation_editor/equation_input.html",
             link: function(scope, element, attrs) {
-                scope.textEntryError = '';
+                scope.textEntryError = [];
```

Comment:
Not sure why this has to be here, but I left it for good measure.

Reply:
Otherwise you got the text `undefined` as the error message the whole time! With your list, I think it wouldn't be an issue, but this is safer still.